### ISSUE TITLE
link opened in new tab by Vimium is not moved to new window

### ISSUE THREAD

**Issue by korakinos** [igorlogius/disable-tabs#2]
**Describe the bug**

When used in conjunction with the [Vimium](https://github.com/philc/vimium/) extension, opening a link using the open-link-in-new-tab shortcut of Vimium does, indeed, open the link in a new tab.

**To Reproduce**
1. Install both disable-tabs and Vimium for Firefox
2. Open a link using the Vimium shortcut 'F'.

**Expected behavior**
disable-tabs should move the tab to a new window.

**Smartphone (please complete the following information):**
 - Device: Pinephone
 - OS: postmarketOS v22.12
 - Browser: Firefox 
 - Version: 102.8.0esr

**Additional context**
This only started happening after the last update. My guess is that it is a side effect of [the commit](https://github.com/igorlogius/disable-tabs/commit/aa5a88c1f03af09657d3b656188bdb571c762714) which fixed [the issue](https://github.com/igorlogius/disable-tabs/issues/1) with Firefox View.

By the way, many thanks for this extension! Very handy on a small screen (phone). (I hide the tab bar with userChrome.css.)

**Comment by korakinos** [comment]
Did you try to open "Create a new account" with `F`? I can confirm that this does not open a new tab but some sort of embedded pop-up (not a OS-level pop-up, something wihin the webpage). (Tested on my laptop. With my phone, because of display size, facebook redirects me to their mobile site, which behaves differently.)

However, there is something special about this button. I don't know much about web development, but I believe it is not a regular link but some sort of javascript thing.

Vimium's  `F` works correctly on standard HTML links, for example any link in [a Wikipedia article](https://en.wikipedia.org/wiki/Firefox).

**Comment by korakinos** [comment]
Wow, I just tested again with `F` on Wikipedia links, and now I am finding that the bug does not occur consistently: Sometimes disable-tabs moves the new tab to a new window, sometimes it doesn't.

**Comment by igorlogius** [comment]
> Did you try to open "Create a new account" with F?

yes, with a capital `f`  aka  `F`  

>  I believe it is not a regular link but some sort of javascript thing.

This could be the explaination ... for Vimiums inconsistent behaviour. 

> Sometimes disable-tabs moves the new tab to a new window, sometimes it doesn't.

Doesnt seem to be an issue with disable-tabs.

There is only one Event Callback (`browser.tabs.onCreated`) in the addon API, which hooks into the tab creation event and that is 100% covered by disable-tabs => ref. https://github.com/igorlogius/disable-tabs/blob/aa5a88c1f03af09657d3b656188bdb571c762714/background.js#L10

At this moment i can only see 3 possibilies: 

1. vimium could also intercepts the onCreated event and somehow "overwrite"/"modify" disable-tabs handling. 
2. vimium could maybe not always create a new tab (bug or intentional) <= could be a javascript issue you mentioned above  
3. firefox could have a bug that causes this strange behaviour

The first 2 options ... can only be confirmed/denied by looking into the Vimium Sources.
The 3rd would ignore for the moment until we can exclude the other 2.

**Comment by korakinos** [comment]
Actually, it turns out, it is not connected to Vimium at all! Even when I open a link in a new tab via the context menu (right click/long tap), disable-tabs *sometimes* moves it into a new tab, sometimes it doesn't. It seems to fail more often when Firefox is busy loading another window/tab.

(Sorry I didn't test this before. It's just not something I usually do on my phone.)

**Comment by igorlogius** [comment]
mmh, i'll try to reproduce it

**Comment by igorlogius** [comment]
i just opened a bunch of tabs around 50 ... by hammering Ctrl+T and all of them were moved into their own window. (with a short delay, as designed)

I am afraid to debug this further i will need some way to be able to redproduce the issue on my side ... and i am drawing a blank at the moment. 

The only idea i still have is to use a proxy to slow down the network and see if that has any influence.

**Comment by korakinos** [comment]
Try holding Ctrl-T instead of hammering. With this I can reproduce it on my laptop.

**Comment by igorlogius** [comment]
yeah, sorry hammering or holding down ... doesnt seem to make a difference.

But my desktop is now litterd with windows 🤣

**Comment by igorlogius** [comment]
Short sitequestion: Are you testing with a new profile?

**Comment by korakinos** [comment]
Just now on my laptop I used a new profile and could reproduce the bug, yes

**Comment by igorlogius** [comment]
i just notice ... that you are using the ESR version ... i'll re-try it with that version and see if that makes a difference.

**Comment by korakinos** [comment]
ESR I am using only on my phone. On my laptop (which I didn't specify in the opening post) I am using 110.0.1. So probably no need to test ESR on your side.

**Comment by korakinos** [comment]
About simulating a slow CPU, this feature appears to be [missing from Firefox dev tools](https://bugzilla.mozilla.org/show_bug.cgi?id=1315658).

If you are on Linux, you could try [cpulimit](https://github.com/opsengine/cpulimit) to throttle the firefox process. I haven't tried this myself.

Or you could just make your CPU very busy with other programs in the background.

**Comment by igorlogius** [comment]
yeah, with the cpulimit lowered i can reproduce the issue ... my first guess would be that this is some kind of fireofx built-in issue ... when limited resources are available to drop cycles for the "addons" ...

**Comment by igorlogius** [comment]
i might have an idea ... that might either circument the issue ... or has no effect at all. 

instead of triggering a detach for each individual tab ... i could try and collect events the event and do the detech when the events have settled.

Might be that this could help ... but might also make no difference. 

Have to make an proof of concept to check.

**Comment by korakinos** [comment]
I see. Thank you for investigating this far!

I already find the detachment delay a little annoying sometimes, so I'm totally with you on not wanting to increase it. 

Note that I originally did not encounter the issue by opening multiple tabs, but just one. Maybe my CPU was busy with some other program at the same time. So only watching out for events within Firefox would probably not be a reliable solution.

Anyway, if I can help by testing anything, let me know!

**Comment by igorlogius** [comment]
If you want, you can check version 1.1.10 and see if that behaves better for you.
 
My inital tests looked promising but having only one datapoint is always a bit of a problem, so i'll be happy to receive any addional feedback.

**Comment by korakinos** [comment]
Interesting! 

On my phone (Firefox ESR) it is definitely not fixed by 1.1.10: I can still sometimes get non-detached tabs by hammering Ctrl-T quickly. However, if I then open one more tab after a few seconds, all the previously non-detached tabs are detached along with the new one.

Success on my laptop, though: There I cannot reproduce the issue anymore, or at least not easily (did not try cpulimit).

**Comment by igorlogius** [comment]
well, sorry i think that is the best i can do for now. 

The real issue doenst seem to be something i can do about. 

I know of another addon, which doesnt have a delay ... i think it's called 'No Tabs' or very similar.  Maybe that works better for your usecase.

**Comment by korakinos** [comment]
No worries, now I can work around the issue by just opening an extra tab when one gets stuck. Thanks a lot for your efforts! (And sorry for spamming your desktop with windows! I do find that quite funny, though.)

Is this worth a bug report to Firefox?

I will also try "No Tabs", hadn't noticed that one before.

**Comment by igorlogius** [comment]
> And sorry for spamming your desktop with windows! I do find that quite funny, though.

No biggy, pkill solved that problem quickly. 

> Is this worth a bug report to Firefox?

Good question ... with resource issue stuff like this it is always a bit tricky to provide a consistent reproducable example ... but might be worth a shot. But i have to admit i would be suprised if the issue would get anything but a low priority.
